cron/centAcl.php

```diff
@@ -220,25 +245,38 @@ function programExit($msg)
          * Add Servicesgroups
          */
         if ($row['all_servicegroups']) {
-            $query = "SELECT sg_id FROM servicegroup WHERE sg_id NOT IN (SELECT DISTINCT sg_id " .
-                "FROM acl_resources_sg_relations WHERE acl_res_id = '" . $row['acl_res_id'] . "')";
-            $res1 = $pearDB->query($query);
+            $res1 = $pearDB->prepare(
+                "SELECT sg_id FROM servicegroup 
+                WHERE sg_id NOT IN (
+                    SELECT DISTINCT sg_id FROM acl_resources_sg_relations
+                    WHERE acl_res_id = :aclResId)"
+            );
+            $res1->bindValue(':aclResId', $row['acl_res_id'], \PDO::PARAM_INT );
+            $res1->execute();
 
             if ($res1->rowCount()) {
                 // set acl_resources.changed flag to 1
                 $aclResourcesUpdated = true;
             }
 
             while ($rowData = $res1->fetch()) {
-                $insert_query = "INSERT INTO acl_resources_sg_relations (sg_id, acl_res_id) VALUES ('" .
-                    $rowData['sg_id'] . "', '" . $row['acl_res_id'] . "')";
-                $pearDB->query($insert_query);
+                $stmt = $pearDB->prepare(
```

Comment:
Don't forget to create a db transaction when there is more than one update request. (=> while)